src/BenchmarksApps/TechEmpower/PlatformBenchmarks/Data/RawDb.cs

```diff
@@ -30,6 +30,15 @@ public RawDb(ConcurrentRandom random, AppSettings appSettings)
             _random = random;
 #if NET7_0_OR_GREATER
             _dataSource = NpgsqlDataSource.Create(appSettings.ConnectionString);
+
+            // For the update benchmark, we use two different connection pools for read/update, to avoid head-of-line
+            // perf issues.
+            var dataSourceBuilder = new NpgsqlDataSourceBuilder(appSettings.ConnectionString);
+
+            dataSourceBuilder.ConnectionStringBuilder.MaxPoolSize = 18;
```

Comment:
Just FYI, this probably doesn't work the way you intended for it to work. Right now we pass `ConnectionStringBuilder` as is to a created data source, which caches it, That means, that by changing any of the settings after calling `Build` you're going to change the cached settings of an already built data source. Concerning `MaxPoolSize`, it works as is because we cache the actual value.

Reply:
Good catch @vonzshik, opened https://github.com/npgsql/npgsql/pull/5000 to track this.